### DISCUSSION TITLE
Merging custom code after Mesh server upgrade

### DISCUSSION THREAD

**Discussion by myintune365** [Ylianst/MeshCentral#6413] [Issues] [answered]
Hi, 
I’ve been customizing the MeshCentral UI and adding features, but I’m encountering issues with merging custom code when upgrading MeshCentral.

I’ve kept most of my changes in separate JavaScript files, but I’ve had to modify meshcore.js to send additional information to the server for a custom scripting output page. This makes upgrading time-consuming due to merging conflicts.

Is there a way to deploy a secondary meshcore.js file to devices, so I can simply reference it in the main file after upgrades instead of manually merging changes each time? Also, are there any recommendations for managing custom code and merging changes more efficiently during upgrades?

Thanks!

**Comment by myintune365** [comment]
Forgot to add, in default.handlebars I renamed a few functions as they were too generic and easy to forgot what they do

For example: 
go() -> goToPage()
Q() -> getById()
QS -> getStyle()
QV - setVisible()

**Comment by si458** [comment, marked as answer]
You can create .js files inside the `modules_meshcore` folder, then require them in the `meshcore.js` file

`require('custom-module');`

Check some of the modules in the folder to understand

**Reply by myintune365** [reply]
So I guess you mean I should create a plugin? I will check some examples, thanks

**Reply by si458** [reply]
Erm yes I suppose so!
As i have seen a few plugins already that do this,
I'm not 100% sure how it all works but I guess trial and error!
